index.ts

```diff
@@ -96,63 +95,59 @@ memoized('bar');
 //=> 2
 ```
 */
-const mem = <
+export default function mem<
 	FunctionToMemoize extends AnyFunction,
-	CacheKeyType
+	CacheKeyType,
 >(
 	fn: FunctionToMemoize,
 	{
 		cacheKey,
 		cache = new Map(),
-		maxAge
-	}: Options<FunctionToMemoize, CacheKeyType> = {}
-): FunctionToMemoize => {
+		maxAge,
+	}: Options<FunctionToMemoize, CacheKeyType> = {},
+): FunctionToMemoize {
 	if (typeof maxAge === 'number') {
-		// TODO: Drop after https://github.com/SamVerschueren/map-age-cleaner/issues/5
-		// @ts-expect-error
-		mapAgeCleaner(cache);
+		mapAgeCleaner(cache as unknown as Map<CacheKeyType, ReturnType<FunctionToMemoize>>);
 	}
 
-	const memoized = function (this: any, ...arguments_) {
-		const key = cacheKey ? cacheKey(arguments_) : arguments_[0];
+	const memoized = function (this: any, ...arguments_: Parameters<FunctionToMemoize>): ReturnType<FunctionToMemoize> {
+		const key = cacheKey ? cacheKey(arguments_) : arguments_[0] as CacheKeyType;
 
 		const cacheItem = cache.get(key);
 		if (cacheItem) {
-			return cacheItem.data;
+			return cacheItem.data; // eslint-disable-line @typescript-eslint/no-unsafe-return
```

Comment:
Because the return value of `AnyFunction` is `any` and `FunctionToMemoize extends AnyFunction`, the linter assumes that `ReturnType<FunctionToMemoize>` is `any`. However, the error is that `any` is **not** assignable to `ReturnType<FunctionToMemoize>`:

>   ✖  118:4  Unsafe return of type any from function with return type ReturnType<FunctionToMemoize>.  @typescript-eslint/no-unsafe-return

This might be a bug with typescript-eslint.